xero/basemanager.py

```diff
@@ -176,6 +176,12 @@ def wrapper(*args, **kwargs):
             if headers is None:
                 headers = {}
 
+            if hasattr(self.credentials, 'tenant_id'):
+                if self.credentials.tenant_id:
+                    headers['Xero-tenant-id'] = self.credentials.tenant_id
+                else:
+                    raise XeroTenantIdNotSet
```

Comment:
I chose hasattr() because if self.credentials is for OAuth1, it will not have the attribute tenant_id. As we should only be raising XeroTenantIdNotSet if self.credentials is for OAuth2, hasattr() was a way of checking which type of credentials were being used.

Would it be better to more explicitly show what is going on, and replace it with the below?
```
if isinstance(self.credentials, OAuth2Credentials):
    if self.credentials.tenant_id:
        headers['Xero-tenant-id'] = self.credentials.tenant_id
    else:
        raise XeroTenantIdNotSet
```

Reply:
I've submitted the above change to the PR.